sdk/compiler/damlc/tests/src/DA/Test/DamlcUpgrades.hs

```diff
@@ -353,18 +358,18 @@ tests damlc =
                       )
                 let sharedDir = dir </> "shared"
                 let sharedDar = sharedDir </> "out.dar"
-                writeFiles sharedDir (projectFile lfVersion ("upgrades-example-" <> name <> "-dep") Nothing Nothing : sharedDepFiles)
```

Comment:
This ugly mess of test code generates daml.yamls and appropriate directories from groups of test files in `test-common/...`. It's messy but ultimately is just handling the different ways we might want to build a series of packages for upgrading.